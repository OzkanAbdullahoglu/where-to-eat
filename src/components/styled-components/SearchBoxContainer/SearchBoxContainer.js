/* eslint-disable linebreak-style */
import styled from 'styled-components';

export const SearchBoxContainer = styled.div`
         position: absolute;
         left: 0px;
         margin: 8px 0 8px 8px;
         top: 0px;
         z-index: 15;
         transition: left 0.5s;
         -webkit-transform: ${(props) =>
    props.closed === true ? 'translateX(-408px)' : 'translateX(0px)'};
         transform: ${(props) =>
    props.closed === true ? 'translateX(-408px)' : 'translateX(0px)'};
         transition-property: -webkit-transform, transform, visibility, opacity;
         transition-duration: 200ms;
         transition-timing-function: cubic-bezier(0.0,0.0,0.2,1);
         visibility: ${(props) => (props.closed === true ? 'hidden' : 'visible')};
       `;

