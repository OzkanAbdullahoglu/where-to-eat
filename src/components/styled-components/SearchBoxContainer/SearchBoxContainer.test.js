/* eslint-disable linebreak-style */
import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import SearchBoxContainer from './SearchBoxContainer';

describe('Search Box Container Render', () => {
  const props = {
    closed: true,
  };

  it('renders search box container', () => {
    const searchBoxWrapper = shallow(<SearchBoxContainer {...props} />);
    expect(toJson(searchBoxWrapper)).toMatchSnapshot();
  });
  it('renders `closed` props', () => {
    const searchBoxWrapper = shallow(<SearchBoxContainer {...props} />);
    expect(searchBoxWrapper).toHaveStyleRule('transform', 'translateX(-408px)');
  });
});
