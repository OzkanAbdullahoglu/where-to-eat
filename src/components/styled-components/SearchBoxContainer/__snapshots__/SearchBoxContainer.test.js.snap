// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box Container Render renders search box container 1`] = `
.c0 {
  position: absolute;
  left: 0px;
  margin: 8px 0 8px 8px;
  top: 0px;
  z-index: 15;
  -webkit-transition: left 0.5s;
  transition: left 0.5s;
  -webkit-transform: translateX(-408px);
  -webkit-transform: translateX(-408px);
  -ms-transform: translateX(-408px);
  transform: translateX(-408px);
  -webkit-transition-property: -webkit-transform,-webkit-transform,visibility,opacity;
  -webkit-transition-property: -webkit-transform,transform,visibility,opacity;
  transition-property: -webkit-transform,transform,visibility,opacity;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0.0,0.0,0.2,1);
  transition-timing-function: cubic-bezier(0.0,0.0,0.2,1);
  visibility: hidden;
}

<div
  className="c0"
/>
`;
